docs/details/other-components/xml.rst

```diff
@@ -0,0 +1,301 @@
+- Animations are not supported yet.
+- Subjects are not supported yet.
+- The documentation is not complete yet.
+
+Main Concept
+~~~~~~~~~~~~
+
+It's important to distinguish between widgets and components:
+
+Widgets are the core building blocks of the UI and are not meant to be loaded at runtime
+but rather compiled into the application. The main characteristics of widgets are:
+
+- Similar to LVGL's built-in widgets.
+- Built from classes.
+- Have a large API with set/get/add/etc. functions.
+- Support "internal widgets" (e.g., tabview's tabs, dropdown's list).
+- Have custom and complex logic inside.
+- Cannot be loaded from XML at runtime because custom code cannot be loaded.
+
+Components are built from other components and widgets and can be loaded at runtime.
+The main characteristics of components are:
+
+- Built from widgets or other components.
+- Can be used for styling widgets.
+- Can contain widgets or other components.
+- Cannot have custom C code.
+- Can be loaded from XML at runtime as they describe only the visuals.
+
+Components
+----------
+
+Overview
+~~~~~~~~
+
+In light of the above, only components can be loaded from XML.
+An example of a ``my_button`` component looks like this:
+
+.. code-block:: xml
+
+    <component>
+        <consts>
+            <px name="size" value="100"/>
+            <color name="orange" value="0xffa020"/>
+        </consts>
+
+        <params>
+            <string name="btn_text"/>
+        </params>
+
+        <styles>
+            <style name="blue" bg_color="0x0000ff" radius="2"/>
+            <style name="red" bg_color="0xff0000"/>
+        </styles>
+
+        <view extends="lv_button" width="#size" styles="blue red:pressed">
+            <my_h3 text="$btn_text" align="center" color="#orange" style_text_color:checked="0x00ff00"/>
+        </view>
+    </component>
+
+- ``<component>``: The root element.
+- ``<consts>``: Constants with ``px`` or ``color`` types.
+Constants can later be referenced as ``#name``.
+- ``<params>``: Parameters with ``int``, ``px``, ``string``, ``color``, or ``style`` types.
+Parameters can later be referenced as ``$name``.
+- ``<styles>``: ``<style>`` properties can be defined with names and properties.
+- ``<view>``: Describes how the component looks. Can reference constants, parameters, and styles.
+
+Naming conventions:
+
+- A standard XML syntax is used.
+- Lowercase letters with ``_`` separation are used for attribute names.
+- The usual variable name rules apply for attribute and tag names: only letters, numbers, `'_'` and can't start with a number.
+- LVGL API is followed as much as possible, e.g., ``align="center"``, ``bg_color="0xff0000"``.
+- ``params`` can be referenced with ``$``
+- ``consts`` can be referenced with ``#``
+- ``styles`` can be attached to states and/or parts like ``styles="red blue:pressed green:focused:scrollbar"``
+- Local styles can be used like  ``<lv_label style_text_color="0xff0000" style_text_color:checked="0x00ff00" ``
```

Comment:
Is `style_` in `style_text_color` a must? Use `text_color` in both C and `<styles/>` would look more unified.

`style_text_color:checked` The `:checked` seems not that common for me, maybe we should only apply local styles to default state and use the full parameter of `<styles/>` if different state is needed?

Reply:
We can consider removing `style_*` from both places but it would be a big API change. Let's treat it seprately.

As LVGL supports local styles for any part and knob it makes sense to me to allow it in XML too. Note that `:checked` can be omitted  to make it affect the main part and default state.